src/main/kotlin/tech/relaycorp/relaynet/ramf/RAMFSerializer.kt

```diff
@@ -0,0 +1,128 @@
+package tech.relaycorp.relaynet.ramf
+
+import com.beanit.jasn1.ber.BerLength
+import com.beanit.jasn1.ber.BerTag
+import com.beanit.jasn1.ber.ReverseByteArrayOutputStream
+import com.beanit.jasn1.ber.types.BerDateTime
+import com.beanit.jasn1.ber.types.BerInteger
+import com.beanit.jasn1.ber.types.BerOctetString
+import com.beanit.jasn1.ber.types.string.BerVisibleString
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+val berDateTimeFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmss")
+
+internal class RAMFSerializer(
+    val concreteMessageType: Byte,
+    val concreteMessageVersion: Byte,
+    val recipient: String,
+    val messageId: String,
+    val creationTimeUtc: LocalDateTime,
+    val ttl: Int,
+    val payload: ByteArray
+) {
+    fun serialize(): ByteArray {
+        val output = ByteArrayOutputStream()
+
+        output.write("Relaynet".toByteArray())
+        output.write(concreteMessageType.toInt())
+        output.write(concreteMessageVersion.toInt())
+        output.write(serializeFields())
+
+        return output.toByteArray()
+    }
+
+    @Throws(IOException::class)
+    private fun serializeFields(): ByteArray {
+        val reverseOS = ReverseByteArrayOutputStream(1000, true)
```

Comment:
The second argument specifies whether the underlying buffer should be resized as the array grows. I couldn't write a test because I didn't know how to spy on it: https://stackoverflow.com/questions/60547003/how-to-spy-on-a-class-constructor-from-an-external-dependency-in-mockk

I'll write that test if/when I get a response.